### PR TITLE
CORE-2634 Connection/reconnection fix.

### PR DESCRIPTION
**Pull request by mikepurvis**
@servos @abencz @efernandez to review.

### CODE REVIEW

**Comment by efernandez** [comment]
:+1:

File: src/sicks3000.cc [hunk 1]
```diff
@@ -123,8 +123,13 @@ int SickS3000::Close()
     // Read controller messages.
     if (serial->ReadPort(cReadBuffer, &read_bytes, 1999) < 0)
```
Comment (abencz) [inline src/sicks3000.cc:124]: Is this return value an `errno`? Could use `strerror()` to print a user friendly version if that's the case: http://beej.us/guide/bgnet/output/html/multipage/perrorman.html

**Comment by abencz** [comment]
LGTM besides the error thing. Would be nice to be able to see why it failed.

File: src/sicks3000.cc (hunk 1 above)
Comment (mikepurvis) [inline src/sicks3000.cc:124]: It's not, but an output occurs in the inner method: https://github.com/clearpathrobotics/s3000_laser/blob/b065680230c0af5d9f04f350b5a7670ab2cc4e4b/external/SerialDevice/src/SerialDevice.cc#L434

I don't like this serial wrapper. I got partway through replacing it with [serial](https://github.com/wjwwood/serial), but I'd rather just rewrite the whole thing as BSD at some point, and then have proper nodelet/laser_pipeline support.

Comment (abencz) [inline src/sicks3000.cc:124]: Oh right, the 500kbit thing :(

Ok, LGTM!

Comment (mikepurvis) [inline src/sicks3000.cc:124]: FWIW, the 500kbit/s was actually a misunderstanding on my part. It _does_ work, and the problem was to do with the self_test method making random extra connections to the serial port and freaking it out.

But by the time I had this all figured out, I'd already written a `select()` blocker for the old serial lib and didn't want to risk even more disruptive changes— as above, my preference would be to just rewrite this some lazy afternoon in the new year.

**Pull request merged**